vector/src/main/java/im/vector/ui/badge/BadgeProxy.kt

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.ui.badge
+
+import android.content.Context
+import android.os.Build
+import im.vector.Matrix
+import me.leolin.shortcutbadger.ShortcutBadger
+import org.matrix.androidsdk.MXDataHandler
+import org.matrix.androidsdk.MXSession
+import org.matrix.androidsdk.util.Log
+import java.util.*
+
+/**
+ * Manage application badge (displayed in the launcher)
+ */
+object BadgeProxy {
+    private const val LOG_TAG = "BadgeProxy"
+
+    /**
+     * Badge is now managed by notification channel, so no need to use compatibility library in recent versions
+     *
+     * @return true if library ShortcutBadger can be used
+     */
+    private fun useShortcutBadger(): Boolean {
+        return Build.VERSION.SDK_INT < Build.VERSION_CODES.O
+    }
+
+    /**
+     * Update the application badge value.
+     *
+     * @param context    the context
+     * @param badgeValue the new badge value
+     */
+    fun updateBadgeCount(context: Context, badgeValue: Int) {
```

Comment:
I think we might need to update notification channels default settings to state if we want to display or not badges (setShowBadge(true/false)).
And will probably need some work on notification builder with .setNumber(messageCount)

Reply:
Yes